From 47d4316480b8a93a7144874e8631ad108955f1ca Mon Sep 17 00:00:00 2001
From: Nic Costa <nic.costa@arm.com>
Date: Thu, 28 Sep 2017 15:36:56 -0500
Subject: [PATCH] change to hacky version of update-client-commit

to avoid the hmac check that is causing the K64F FOTA failure

Signed-off-by: Nic Costa <nic.costa@arm.com>
---
 update-client-common.lib | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/update-client-common.lib b/update-client-common.lib
index 37a924c..8ad91d2 100644
--- a/update-client-common.lib
+++ b/update-client-common.lib
@@ -1 +1 @@
-https://github.com/ARMmbed/update-client-common/#3949b0af706aa487a39fcfc7f03f912d2cf35fab
+https://github.com/costanic/update-client-common/#e8c3d9f5e31ca476cbca158d99fb2b0fe16d2126
-- 
2.14.1

