diff --git a/targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/TARGET_MCU_K64F/device/TOOLCHAIN_GCC_ARM/MK64FN1M0xxx12.ld b/targets/TARGET_Freescale/TARGET_MCUXpr
index 327e9de..41ad57d 100644
--- a/targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/TARGET_MCU_K64F/device/TOOLCHAIN_GCC_ARM/MK64FN1M0xxx12.ld
+++ b/targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/TARGET_MCU_K64F/device/TOOLCHAIN_GCC_ARM/MK64FN1M0xxx12.ld
@@ -64,7 +64,7 @@ __stack_size__ = 0x400;
 __heap_size__ = 0x6000;

 #if !defined(MBED_APP_START)
-  #define MBED_APP_START 0
+  #define MBED_APP_START 0x20400
 #endif

 #if !defined(MBED_APP_SIZE)
